service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/SpecimenMapper.java

```diff
@@ -0,0 +1,239 @@
+        if (specimen.hasCollection() && specimen.getCollection().hasCollector()) {
+            Reference collector = specimen.getCollection().getCollector();
+
+            return Optional.of(messageContext.getIdMapper().get(collector));
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<String> buildAccessionIdentifier(Specimen specimen) {
+        if (specimen.hasAccessionIdentifier() && specimen.getAccessionIdentifier().hasValue()) {
+            return Optional.of(specimen.getAccessionIdentifier().getValue());
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<String> buildType(Specimen specimen) {
+        if (specimen.hasType()) {
+            return CodeableConceptMappingUtils.extractTextOrCoding(specimen.getType());
+        }
+
+        return Optional.empty();
+    }
+
+    private String mapObservationsAssociatedWithSpecimen(Specimen specimen, List<Observation> observations) {
+        List<Observation> observationsAssociatedWithSpecimen = observations.stream()
+            .filter(Observation::hasSpecimen)
+            .filter(observation -> observation.getSpecimen().getReference().equals(specimen.getId()))
+            .collect(Collectors.toList());
+
+        return observationsAssociatedWithSpecimen.stream().map(
+            observationAssociatedWithSpecimen -> observationMapper.mapObservationToCompoundStatement(
+                observationAssociatedWithSpecimen,
+                observations
+            ))
+            .collect(Collectors.joining());
+    }
+
+    private Optional<String> buildPertinentInformation(Specimen specimen) {
+        PertinentInformationBuilder pertinentInformationBuilder = new PertinentInformationBuilder();
+        if (specimen.hasCollection()) {
+            Specimen.SpecimenCollectionComponent collection = specimen.getCollection();
+
+            collection.getExtensionsByUrl(FASTING_STATUS_URL).stream().findFirst()
+                .ifPresent(extension -> {
+                    Type value = extension.getValue();
+                    if (value instanceof CodeableConcept) {
+                        pertinentInformationBuilder.fastingStatus((CodeableConcept) value);
+                    } else if (value instanceof Duration) {
+                        pertinentInformationBuilder.fastingDuration((Duration) value);
+                    }
+                });
+
+            if (collection.hasQuantity()) {
+                pertinentInformationBuilder.quantity(collection.getQuantity());
+            }
+
+            if (collection.hasBodySite()) {
+                pertinentInformationBuilder.collectionSite(collection.getBodySite());
+            }
+        }
+
+        specimen.getNote().stream()
+            .map(Annotation::getText)
+            .forEach(pertinentInformationBuilder::note);
+
+        return getEffectiveTime(specimen)
+            .map(pertinentInformationBuilder::build)
+            .orElseGet(pertinentInformationBuilder::build);
```

Comment:
This threw me! Could we perhaps rename the first one to `buildWithDateTime` or some such?

Reply:
Updated in 4fcec5ecc2c51dffcdf323fad657adfbb0d2a555